cmd/kubeadm/app/phases/upgrade/postupgrade_test.go

```diff
@@ -101,3 +104,98 @@ func TestRollbackFiles(t *testing.T) {
 		t.Fatalf("Expected error contains %q, got %v", errString, err)
 	}
 }
+
+func TestWriteKubeletConfigFiles(t *testing.T) {
```

Comment:
```
--- FAIL: TestWriteKubeletConfigFiles (0.00s)
    /Users/pacoxu/git/gopath/src/k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade/postupgrade_test.go:155: Expected error contains "no kubelet component config found.*no such file or directory", got failed to create directory "/etc/kubernetes/tmp": mkdir /etc/kubernetes: permission denied
FAIL
FAIL	k8s.io/kubernetes/cmd/kubeadm/app/phases/upgrade	1.058s
FAIL
```
This UT will fail on Mac for permission.

Reply:
If this is a problem on the Mac, it should be a long-standing issues, as the `CreateTempDirForKubeadm` is called in many places. 

Have you run the test with `root`? 

Let me find a Mac and try it out, anyway, if it is an issue, the issue should be fixed separately.